extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/StartupBuildSteps.java

```diff
@@ -51,26 +48,15 @@
     private static final Logger LOGGER = Logger.getLogger(StartupBuildSteps.class);
 
     @BuildStep
-    AnnotationsTransformerBuildItem annotationTransformer(CustomScopeAnnotationsBuildItem customScopes) {
-        return new AnnotationsTransformerBuildItem(new AnnotationsTransformer() {
-
-            @Override
-            public boolean appliesTo(org.jboss.jandex.AnnotationTarget.Kind kind) {
-                return kind == org.jboss.jandex.AnnotationTarget.Kind.CLASS;
-            }
-
-            @Override
-            public void transform(TransformationContext context) {
-                if (context.isClass() && !customScopes.isScopeDeclaredOn(context.getTarget().asClass())) {
-                    // Class with no built-in scope annotation but with @Scheduled method
-                    if (Annotations.contains(context.getTarget().asClass().classAnnotations(), STARTUP_NAME)) {
-                        LOGGER.debugf("Found @Startup on a class %s with no scope annotations - adding @ApplicationScoped",
-                                context.getTarget());
-                        context.transform().add(ApplicationScoped.class).done();
-                    }
-                }
-            }
-        });
+    AutoAddScopeBuildItem addScope(CustomScopeAnnotationsBuildItem customScopes) {
+        // Class with no built-in scope annotation but with @Startup method
+        return AutoAddScopeBuildItem.builder()
+                .defaultScope(BuiltinScope.APPLICATION)
+                .match((clazz, annotations, index) -> clazz.classAnnotation(STARTUP_NAME) != null
+                        && !customScopes.isScopeDeclaredOn(clazz))
```

Comment:
This rule is applied automatically: https://github.com/quarkusio/quarkus/blob/master/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/AutoAddScopeProcessor.java#L51